### DISCUSSION TITLE
How do I generate a UUID in Micropython?

### DISCUSSION THREAD

**Discussion by Razboinicul** [micropython/micropython#15728] [Libraries & Drivers]
I am trying to generate a UUID automatically on my ESP32.

**Comment by shariltumin** [comment]
Perhaps you can use the following?
```py
>>> import machine
>>> machine.unique_id().hex()
```

**Comment by mattytrentini** [comment]
I have an outstanding PR https://github.com/micropython/micropython-lib/pull/504 that can generate version 1,3, 4 and 5 UUIDs. It still need to be tidied up before merging but it's probably usable as-is.
